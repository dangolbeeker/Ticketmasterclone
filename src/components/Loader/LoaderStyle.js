// Packages
import styled, { keyframes } from 'styled-components';


export const rotate = keyframes`
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
`

export const LoaderItem = styled.div`
    margin: 0 auto;
    vertical-align: center;
    border: 8px solid #f3f3f3;
    border-radius: 50%;
    border-top: 8px solid #3498db;
    width: 60px;
    height: 60px;
    -webkit-animation: ${rotate} 2s linear infinite; /* Safari */
    animation: ${rotate} 2s linear infinite;
`
