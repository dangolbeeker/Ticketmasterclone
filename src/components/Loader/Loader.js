// Packages
import React from "react";

// Styles
import * as Style from './LoaderStyle.js';


class Loader extends React.Component {
  render() {
    return(
        <div>
            <Style.LoaderItem></Style.LoaderItem>
        </div>
    );
  }
}


export default Loader;
