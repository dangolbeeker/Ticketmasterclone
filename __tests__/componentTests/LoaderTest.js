// Packages
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

// Components
import Loader from '../../src/components/Loader/Loader';


configure({ adapter: new Adapter() });

describe('Loader Component', () => {
    it('Should render without errors', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper.exists()).toBe(true)
    });

    test('Snaphot testing', () => {
        const wrapper = renderer.create(<Loader />).toJSON();
        expect(wrapper).toMatchSnapshot();
    })
})
