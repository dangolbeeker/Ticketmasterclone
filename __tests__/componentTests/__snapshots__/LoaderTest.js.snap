// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Snaphot testing 1`] = `
<div>
  <div
    className="sc-bdVaJa jPSyVL"
  />
</div>
`;
